clippy_lints/src/copies.rs

```diff
@@ -315,6 +315,21 @@ struct BlockEqual {
     expr_eq: bool,
 }
 
+fn block_contains_todo_macro(cx: &LateContext<'_>, block: &Block<'_>) -> bool {
```

Comment:
You're using the span of the block there, which isn't created by the macro. You'll want to use either the final expression or statement depending on whether `{ todo!() }` or `{ todo!; }` was typed.

For the macro call site, you'll only need to worry about that if the `todo` macro calls a different macro for it's implementation. I haven't checked but it might be using `panic` internally.

Reply:
Alright. How do I get the final expression and the final statement?

Reply:
`block.expr` is the final expression if it exists. `block.stmts` is all the statements in the bock.

If you're unsure what you can do with a type just look it up in the docs. https://doc.rust-lang.org/nightly/nightly-rustc/